lib/cocoapods-core/source.rb

```diff
@@ -201,7 +201,8 @@ def search(query)
       if query.is_a?(Dependency)
         query = query.root_name
       end
-      if (specs_dir + query).directory?
+      spec_dir = specs_dir + query
+      if Pathname.glob(spec_dir) == [spec_dir]
```

Comment:
I don't think we need to, as there are no valid spec names that contain glob characters